parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -1097,10 +1097,26 @@ prettyParseError ::
   String ->
   Parser.Err v ->
   Pretty ColorText
-prettyParseError s = \case
-  P.TrivialError _ (LexerError ts e) _ -> go e
+prettyParseError s e =
+  mconcat (fst <$> renderParseErrors s e) <> lexerOutput
+  where
+    lexerOutput :: Pretty (AnnotatedText a)
+    lexerOutput =
+      if showLexerOutput
+        then "\nLexer output:\n" <> fromString (L.debugLex' s)
+        else mempty
+
+renderParseErrors ::
```

Comment:
We need the range affected by each error for making diagnostics, this was a huge pain, but I think it makes sense to keep the ranges alongside the error messages here, also this ensures they'll be updated as new error messages are added, or old ones are edited.